### PR TITLE
Added HSV color support from FastLED and multiple effects

### PR DESCRIPTION
**Pull request by CedArctic**
**Changes:**
- Introduced modified HSV color function from FastLED
- Added Breathing & Spectrum lighting effects
- Added Rainbow Flow & Rainbow Waterfall effects

### CODE REVIEW

File: source/miniFastLED.c [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+#define HSV_SECTION_6 (0x20)
+#define HSV_SECTION_3 (0x40)
+
+// Shared array used to place in the results of hsv2rgb
+uint8_t rgbArray[3];
+
+// Convert HSV to RGB
+void hsv2rgb(uint8_t hue, uint8_t sat, uint8_t val, uint8_t* rgbResults){
+
+        // Convert hue, saturation and brightness ( HSV/HSB ) to RGB
```
Comment (Codetector1374) [inline source/miniFastLED.c:20]: Reindent

File: main.c [hunk 2]
```diff
@@ -203,6 +242,54 @@ void animationCallback(GPTDriver* _driver){
...
+    // Breathing
+    case LEN(colorPalette) + 4:
+      gptChangeInterval(_driver, ANIMATION_TIMER_FREQUENCY/30);
+      setAllKeysColorHSV(ledColors, 85, 255, value);
+      if(value >= 180){
+        direction = -1;
+      }else if(value <= 0){
+        direction = 1;
+      }
+      value+=direction;
```
Comment (Codetector1374) [inline main.c:255]: Space around =

File: main.c [hunk 3]
```diff
@@ -203,6 +242,54 @@ void animationCallback(GPTDriver* _driver){
...
+    // Spectrum
+    case LEN(colorPalette) + 5:
+      gptChangeInterval(_driver, ANIMATION_TIMER_FREQUENCY/15);
+      setAllKeysColorHSV(ledColors, value, 255, 125);
+      if(value >= 179){
+        direction = -1;
+      }else if(value <= 1){
+        direction = 1;
+      }
+      value+=direction;
```
Comment (Codetector1374) [inline main.c:267]: Space

**Comment by CedArctic** [comment]
@Codetector1374 I have indeed tested and have been running the new effects for a few days now. My latest commit addresses the styling changes. If anything else needs to be done, let me know.

**Pull request merged**